### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by no4kar**
[DEMO LINK](https://no4kar.github.io/react_todo-app-with-api/)

### CODE REVIEW

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your page
This button doesn't work
image

File: src/App.tsx [hunk 1]
```diff
@@ -1,24 +1,174 @@
...
-    return <UserWarning />;
-  }
+  // #region STATE
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMsg, setErrorMsg] = useState<TodoError>(TodoError.NONE);
+  const [filter, setFilter] = useState<Filter>(Filter.ALL);
+  const [processings, setProcessings] = useState<number[]>([]);
+  // #endregion
```
Comment (volodymyr-soltys97) [inline src/App.tsx:26]: Remove comments everywhere
```suggestion
```

File: src/App.tsx [hunk 2]
```diff
@@ -1,24 +1,174 @@
...
+                />
+              )}
+            </section>
+
+            <TodoFooter
+              filter={filter}
+              onFilterChange={handleFilterChange}
+              onClearCompleted={deleteCompletedTodos}
+              quantityActiveTodos={activeTodos.length}
+              isAnyTodoComplete={todos.some(todo => todo.completed)}
```
Comment (volodymyr-soltys97) [inline src/App.tsx:160]: Move this logic to the helper function and use it here

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Great!
Let's improve your code

**Review by anastasiia-tilikina — CHANGES_REQUESTED** [review]
Well done! We're almost here)

just one more minor fix.

these actions should always be in the center, even if 'clear completed' button is not shown.

image

**Review by xiimiik — APPROVED** [review]
Nice work!

**Pull request closed without merging**